packages/core/src/storage/compare.ts

```diff
@@ -362,7 +362,15 @@ export class StorageLayoutComparator {
       }
 
       case 't_userDefinedValueType': {
-        if (original.item.numberOfBytes === undefined || updated.item.numberOfBytes === undefined) {
+        const underlyingMatch =
+          original.item.underlying &&
+          updated.item.underlying &&
+          original.item.underlying.id == updated.item.underlying.id;
```

Comment:
Let's prefer strict equals since the types should match in this case:
```suggestion
          original.item.underlying.id === updated.item.underlying.id;
```